File: springboot-backend/src/main/java/com/wedspot/backend/services/implementation/DashboardService.java
package com.wedspot.backend.services.implementation;

import com.wedspot.backend.services.IDashboardService;
import jakarta.persistence.EntityManager;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.HashMap;
import java.util.Map;

@Service
@RequiredArgsConstructor
public class DashboardService implements IDashboardService {

    private final EntityManager entityManager;

    @Override
    public Map<String, Object> getDashboardStats() {
        Map<String, Object> stats = new HashMap<>();
        
        try {
            Long totalVendors = (Long) entityManager.createNativeQuery("SELECT COUNT(*) FROM vendors").getSingleResult();
            Long totalBookings = (Long) entityManager.createNativeQuery("SELECT COUNT(*) FROM bookings").getSingleResult();
            Long totalUsers = (Long) entityManager.createNativeQuery("SELECT COUNT(*) FROM users").getSingleResult();
            
            stats.put("totalVendors", totalVendors);
            stats.put("totalBookings", totalBookings);
            stats.put("totalUsers", totalUsers);
            stats.put("revenue", 4520000); // Mock revenue logic for now
        } catch (Exception e) {
            stats.put("error", "Could not fetch stats: " + e.getMessage());
        }

        return stats;
    }
}

File: backend/Services/Implementations/DashboardService.cs
using backend.Repositories;

namespace backend.Services
{
    public class DashboardService : IDashboardService
    {
        private readonly IUserRepository _userRepository;
        private readonly IVendorRepository _vendorRepository;
        private readonly IBookingRepository _bookingRepository;

        public DashboardService(IUserRepository userRepository, IVendorRepository vendorRepository, IBookingRepository bookingRepository)
        {
            _userRepository = userRepository;
            _vendorRepository = vendorRepository;
            _bookingRepository = bookingRepository;
        }

        public async Task<Dictionary<string, object>> GetDashboardStats()
        {
            var users = await _userRepository.GetAllAsync();
            var vendors = await _vendorRepository.GetAllAsync();
            var bookings = await _bookingRepository.GetAllAsync();

            return new Dictionary<string, object>
            {
                { "totalUsers", users.Count },
                { "totalVendors", vendors.Count },
                { "totalBookings", bookings.Count },
                { "revenue", 15000.50 } // Dummy revenue
            };
        }
    }
}
